src/org/impact2585/frc2016/input/InputMethod.java

```diff
@@ -1,85 +1,148 @@
 package org.impact2585.frc2016.input;
 
 /**
- * Classes implement this if they are a type of input
+ * Classes extend this if they are a type of input
  */
-public interface InputMethod {
-	
+public abstract class InputMethod {
+
 	/**
 	 * @returns how far the robot should move
 	 */
-	public default double forwardMovement() {
+	public double forwardMovement() {
 		return 0;
 	}
-	
+
 	/**
-	 * @returns how much the robot should robot
+	 * @returns how much the robot should rotate
 	 */
-	public default double rotationValue() {
+	public double rotationValue() {
 		return 0;
 	}
-	
+
 	/**
-	 * @returns true if the robot drivetrain should invert
+	 * @returns true if the robot's drivetrain should invert
 	 */
-	public default boolean invert() {
+	public boolean invert() {
 		return false;
 	}
-	
+
+
+	/**
+	 * @returns 1 or -1 if the input is digital, but can also return the analog input if the input is analog
+	 */
+	public double moveTopArm() {
+		if(digitalTopArmForward() && !digitalTopArmBackward()) {
+			return 1;
+		} else if (!digitalTopArmForward() && digitalTopArmBackward()) {
+			return -1; 
+		} else if (analogTopArmForward() > 0 && analogTopArmBackward() == 0) {
+			return analogTopArmForward();
+		} else if (analogTopArmForward() == 0 && analogTopArmBackward() > 0) {
+			return analogTopArmBackward();
```

Comment:
Check if the sign of that is right. We really need to add to our unit tests so that they catch more mistakes.